### ISSUE TITLE
character shaking when typing Chinese characters

### ISSUE THREAD

**Issue by zwyyy456** [microsoft/vscode#212779]
Type: <b>Bug</b>

My VS Code has the `vscode.neovim` plugin installed, and when I type Chinese characters in insert mode, the Chinese characters in the line I'm typing shake rapidly, which is quite uncomfortable to watch. This issue only occurs on macOS and not on Windows. I have reported this problem to the developers of vscode.neovim, and they said it should be a bug in VS Code and have closed the corresponding issue. This bug appears after version 1.10.1 of `vscode.neovim`.

VS Code version: Code 1.89.1 (dc96b837cf6bb4af9cd736aa3af08cf8279f7685, 2024-05-07T05:14:32.757Z)
OS version: Darwin arm64 23.4.0
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 Pro (8 x 24)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|5, 4, 5|
|Memory (System)|16.00GB (0.39GB free)|
|Process Argv|--crash-reporter-id 85b42040-3384-4b8a-9359-d8ca9a96eb2b|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (54)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-neovim|asv|1.10.1
doxdocgen|csc|1.4.0
fastcompare|Dav|1.2.1
EditorConfig|Edi|0.16.4
outline-map|Ger|1.4.0
vscode-pull-request-github|Git|0.88.0
go|gol|0.41.4
latex-workshop|Jam|9.20.1
plantuml|jeb|2.18.1
better-cpp-syntax|jef|1.17.2
cmake-language-support-vscode|jos|0.0.9
vscode-gist|ken|3.0.3
beancount|Len|0.11.0
vscode-clangd|llv|0.1.28
asm-code-lens|maz|2.6.0
rainbow-csv|mec|3.11.0
vscode-docker|ms-|1.29.1
vscode-dotnet-runtime|ms-|2.0.3
black-formatter|ms-|2024.2.0
debugpy|ms-|2024.6.0
isort|ms-|2023.10.1
python|ms-|2024.6.0
vscode-pylance|ms-|2024.5.1
jupyter|ms-|2024.4.0
jupyter-keymap|ms-|1.1.2
jupyter-renderers|ms-|1.0.17
vscode-jupyter-cell-tags|ms-|0.1.9
vscode-jupyter-slideshow|ms-|0.1.6
remote-containers|ms-|0.362.0
remote-ssh|ms-|0.110.1
remote-ssh-edit|ms-|0.86.0
remote-wsl|ms-|0.88.2
cmake-tools|ms-|1.17.17
cpptools-extension-pack|ms-|1.3.0
powershell|ms-|2024.2.1
remote-explorer|ms-|0.4.3
remote-server|ms-|1.5.1
java|red|1.30.0
vscode-yaml|red|1.14.0
partial-diff|ryu|1.4.3
markdown-preview-enhanced|shd|0.8.13
vs-picgo|Spa|2.1.6
even-better-toml|tam|0.19.2
xmake-vscode|tbo|2.3.5
cmake|twx|0.0.17
vscode-lldb|vad|1.10.0
intellicode-api-usage-examples|Vis|0.2.8
vscodeintellicode|Vis|1.3.1
vscode-java-debug|vsc|0.57.0
vscode-java-dependency|vsc|0.23.6
vscode-java-pack|vsc|0.26.0
vscode-java-test|vsc|0.41.1
vscode-maven|vsc|0.44.0
vscode-icons|vsc|12.7.0

(2 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vspor879:30202332
vspor708:30202333
vspor363:30204092
tftest:31042121
vscoreces:30445986
vscod805cf:30301675
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
vscrp:30673768
2i9eh265:30646982
962ge761:30959799
pythongtdpath:30769146
welcomedialog:30910333
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythontestfixt:30902429
pythonregdiag2:30936856
pythonmypyd1:30879173
pythoncet0:30885854
h48ei257:31000450
pythontbext0:30879054
accentitlementsc:30995553
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
showvideoc:31016891
chatpanelc:31048052
dsvsc021:30996838
724cj586:31013169
fchga941:31046352
pythonprc:31047982
dwnewjupyter:31046869

```

</details>

**Comment by alexdima** [comment]
@zwyyy456 Can you please create a screen recording showing the problem? Does the same issue occur after disabling the `vscode.neovim` extension?

**Comment by zwyyy456** [comment]
It won't disppear when I disable the `vscode.neovim`, I have report this problem to the developers of `vscode.neovim`, but they said it's a problem about vscode and close the issue.


https://github.com/microsoft/vscode/assets/45388797/38dc9fc1-d841-4c0e-8f54-fd99b2dd9497



Here are some issues about this problem:

https://github.com/microsoft/vscode/issues/211856
https://github.com/vscode-neovim/vscode-neovim/issues/1921

**Comment by aiday-mar** [comment]
Hi I am trying to reproduce this issue but am having a bit of trouble as I don't speak Chinese. You mention I should be able to reproduce this issue without the VS Code Neovim extension installed. First of all, do you still see this issue? If yes could you tell me what keyboard you are using and what exactly you are typing so I can reproduce this issue? Thank you.

**Comment by zwyyy456** [comment]
What I mean is that when I disable the `vscode.neovim` extension, this bug no longer occurs, but when I enable the `vscode.neovim` extension, the bug related to Chinese input appears. This seems to be a bug in the `vscode.neovim` extension. So, I consulted the developer of the `vscode.neovim` extension, and their response was that this is actually a bug caused by VSCode itself.

> First of all, this issue is actually a problem with vscode. Now vscode-neovim uses default:type correctly. 

> Currently, only users who need to use composite keys and input CJK characters may encounter this issue. (And just found out that similar issues existed before.) The composite keys refers to the mappings in insert mode, typically 'jj' or 'jk', to return to normal mode.

see [vscode.neovim issues](https://github.com/vscode-neovim/vscode-neovim/issues/1921)

**Comment by aiday-mar** [comment]
Currently I am trying to reproduce this issue but I am having trouble configuring neovim on my macos. I get the following error 'neovim gets disconnected'. After searching on the neovim repo it would seem that other people are encountering this issue here https://github.com/vscode-neovim/vscode-neovim/issues/2165. Seems like an upstream issue which is preventing from reproducing the bug. 

https://github.com/user-attachments/assets/b6cf5d71-f93d-496c-8271-a7aeb6d5bc0f

**Comment by xiyaowong** [comment]
@aiday-mar Hi, I'm one of the maintainers of `vscode-neovim` and noticed that you are working on this issue. I believe this issue is consistent with #211856.

**Comment by aiday-mar** [comment]
Hi thank you for filing this issue. I was able to reproduce this issue with an extension that overrides the _type_ and `replacePreviousChar` commands as in https://github.com/microsoft/vscode/issues/211856, and I verified that this jittering does not happen with the setting `editor.experimentalEditContextEnabled`. We will turn on the experimental edit context on by default soon, so I will close this issue. 

https://github.com/user-attachments/assets/6f730c2e-fdef-4618-a3c6-9dfd52fe1fd6
